File: solutions/zig/isogram/1/isogram.zig
pub fn isIsogram(str: []const u8) bool {
    var letters: [26]u8 = .{0} ** 26;

    for (s) |value| {
        if (!std.ascii.isAlphabetic(value)) {
            continue;
        }

        const index = std.ascii.toLower(value) - 'a';
        if (letters[index] > 1) {
            return true;
        }
    }

    return false;
}

File: solutions/go/isogram/1/isogram.go
package isogram

import "strings"

func IsIsogram(word string) bool {
	var letters = make(map[rune]bool)

	for _, l := range strings.ToLower(word) {
		if l < 'a' || l > 'z' {
			continue
		}

		if _, ok := letters[l]; !ok {

			letters[l] = true
		} else {
			return false
		}
	}
	return true
}
